common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -58,9 +69,15 @@ def connect_to_mongodb(
             wait_time=retry_wait_time
         )
 
+    # default the authSource to be whatever db we are connecting to (for backwards compatiblity)
+    authSource=db
```

Comment:
No need to mention Atlas in the code. The commit or pull request are great places to explain.

```suggestion
    # Allow customizing the authSource to be more compatible with different MongoDB providers.        
    authSource = kwargs.get('authSource', db)
```